Examples/Sources/MacroExamples/Implementation/Peer/AddCompletionHandlerMacro.swift

```diff
@@ -25,20 +25,22 @@ public struct AddCompletionHandlerMacro: PeerMacro {
   ) throws -> [DeclSyntax] {
     // Only on functions at the moment. We could handle initializers as well
     // with a bit of work.
-    guard let funcDecl = declaration.as(FunctionDeclSyntax.self) else {
+    guard var funcDecl = declaration.as(FunctionDeclSyntax.self) else {
       throw CustomError.message("@addCompletionHandler only works on functions")
     }
 
     // This only makes sense for async functions.
     if funcDecl.signature.effectSpecifiers?.asyncSpecifier == nil {
       let newEffects: FunctionEffectSpecifiersSyntax
-      if let existingEffects = funcDecl.signature.effectSpecifiers {
-        newEffects = existingEffects.with(\.asyncSpecifier, .keyword(.async))
+      if var existingEffects = funcDecl.signature.effectSpecifiers {
+        existingEffects.asyncSpecifier = .keyword(.async)
+        newEffects = existingEffects
```

Comment:
I would flip this, so we are modifying the new effects because technically we can’t really modify the existing effects (they are the old ones after all).

```suggestion
        newEffects = existingEffects
        newEffects.asyncSpecifier = .keyword(.async)
```

Reply:
Mea culpa 😅 that definitely makes more sense!